applications/ecallmgr/src/ecallmgr_fs_fetch_directory.erl

```diff
@@ -0,0 +1,204 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2011-2017, 2600Hz INC
+%%% @doc
+%%% Directory lookups from FS
+%%% @end
+%%% @contributors
+%%%   James Aimonetti
+%%%   Karl Anderson
+%%%-------------------------------------------------------------------
+-module(ecallmgr_fs_fetch_directory).
+
+-export([directory_lookup/1]).
+-export([lookup_user/4]).
+-export([init/0]).
+
+-include("ecallmgr.hrl").
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec init() -> 'ok'.
+init() ->
+    kazoo_bindings:bind(<<"fetch.directory.*.*.*">>, ?MODULE, 'directory_lookup'),
+    'ok'.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+-spec directory_lookup(map()) -> fs_handlecall_ret().
+directory_lookup(#{node := Node, fetch_id := FetchId, payload := JObj}) ->
+    kz_util:put_callid(FetchId),
+    lager:debug("received fetch request (~s) user directory from ~s", [FetchId, Node]),
+    case kzd_fetch:fetch_action(JObj, <<"sip_auth">>) of
+        <<"reverse-auth-lookup">> -> lookup_user(Node, FetchId, <<"reverse-lookup">>, JObj);
```

Comment:
`lookup_user/4` is passed a JObj here but in the code below, it calls it Props and uses `props:get_*`

Reply:
it was the kzd_freeswitch mix